components/automate-ui/src/app/services/http/http-client-auth.interceptor.ts

```diff
@@ -1,26 +1,44 @@
-import { throwError as observableThrowError,  Observable } from 'rxjs';
-
-import { catchError } from 'rxjs/operators';
-import { Injectable } from '@angular/core';
 import { HttpEvent, HttpHandler, HttpInterceptor, HttpRequest } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Store } from '@ngrx/store';
+import { throwError as observableThrowError, Observable } from 'rxjs';
+import { catchError } from 'rxjs/operators';
 import { get } from 'lodash/fp';
-import { ChefSessionService } from '../chef-session/chef-session.service';
+
+import { NgrxStateAtom } from 'app/ngrx.reducers';
+import { ChefSessionService } from 'app/services/chef-session/chef-session.service';
+import * as selectors from 'app/services/projects-filter/projects-filter.selectors';
+import { ProjectsFilterOption } from '../projects-filter/projects-filter.reducer';
 
 @Injectable()
 export class HttpClientAuthInterceptor implements HttpInterceptor {
 
-  constructor(private chefSession: ChefSessionService) {}
+  private projects: string;
+
+  constructor(
+    private chefSession: ChefSessionService,
+    private store: Store<NgrxStateAtom>
+  ) {
+    this.store.select(selectors.options)
+      .subscribe((options: ProjectsFilterOption[]) => {
+        this.projects = options.filter(p => p.checked).map(p => p.value).join(', ');
+      });
+  }
 
   intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+    let headers = request.headers.set('Authorization', `Bearer ${this.chefSession.id_token}`);
+    if (this.projects) {
```

Comment:
Important to only send when there are some; otherwise we are making an invalid request and get a 403, because all projects must have non-zero length.